ext/storage_sources/azure_store/azure_connection.h

```diff
@@ -0,0 +1,24 @@
+#ifndef AZURECONNECTION
+#define AZURECONNECTION
+
+// Includes necessary for connection
+#include <azure/core.hpp>
+#include <azure/storage/blobs.hpp>
+
+#include <string>
+
+class azure_connection {
+    public:
+    azure_connection(const std::string &bucket_name, const std::string &obj_prefix = "");
+    int list_objects(std::vector<std::string> &objects) const;
+    int put_object(const std::string &file_name) const;
+    int delete_object() const;
+    int get_object(const std::string &path) const;
```

Comment:
Are you sure all of these are const? Stuff like `put_object` seems like it would change at least some internal state.

Reply:
The put_object function is just supposed to upload an object in the container. We're not supposed to be making any internal modifications to objects in the cloud. Could you please specify what you mean by internal state?

Reply:
Ah OK. I thought class might end up tracking (say) what's been uploaded or not, which is what I meant by "internal state". No harm in removing the `const` later, so I'll resolve this - thanks for the explainer!